compiler/rustc_hir/src/hir.rs

```diff
@@ -228,13 +228,54 @@ impl<'hir> PathSegment<'hir> {
     }
 }
 
+/// A constant that enters the type system, e.g. through const generics or even
+/// array lengths.
+///
+/// These are distinct from [`AnonConst`] in part because with the  plan for
+/// `min_generic_const_args`, arbitrary anonymous constants (like `Foo<{N + 1}>`)
+/// will *not* be allowed to use generic parameters. Instead, it will be necessary
+/// to add indirection using a free constant that itself has const parameters.
+///
+/// So, `ConstArg` (specifically, [`ConstArgKind`]) distinguishes between const args
+/// that are [just paths](ConstArgKind::Path) (currently just bare const params)
+/// versus const args that are literals or have arbitrary computations (e.g., `{ 1 + 3}`).
```

Comment:
I think the primary explanation here should not be about `min_generic_const_args` as even if that feature doesn't wind up working out we'll still want this PR.

```suggestion
/// A constant that enters the type system, used for arguments to const generics (e.g. array lengths).
///
/// These are distinct from [`AnonConst`] as anon consts in the type system are not allowed
/// to use any generic parameters, therefore we must represent `N` differently. Additionally
/// future designs for supporting generic parameters in const arguments will likely not use
/// an anon const based design. 
///
/// So, `ConstArg` (specifically, [`ConstArgKind`]) distinguishes between const args
/// that are [just paths](ConstArgKind::Path) (currently just bare const params)
/// versus const args that are literals or have arbitrary computations (e.g., `{ 1 + 3}`).
```

Reply:
Sounds good! I applied your suggestion (with an edit to fix a missing space from the original version).